src/main/java/io/stargate/sgv2/jsonapi/service/resolver/matcher/FilterMatchRule.java

```diff
@@ -30,7 +30,7 @@ public record FilterMatchRule<T extends Command & Filterable>(
   @Override
   public Optional<DBLogicalExpression> apply(CommandContext commandContext, T command) {
     return matcher
-        .apply(command)
+        .apply(commandContext, command)
```

Comment:
Hey stupid idea time - do we need an object for CommandWithContext ?

Reply:
Hmmh. That might be a good idea, not sure. But it'd be more... "Command and context" wrapper, I think?
Or we could make either Command refer to context (needs to assign after construction), or vice versa (context referencing command).

Main benefit being we can pass just one thing around.

Good question, not sure what the answer is. :)

Reply:
ok, lets not add to this PR for now. I think I would lean more towards adding Context to the Command , but the command is created before the context so to be pragmatic we may need to add the Command to the CommandContext. 

It will be a small change but hit a bunch of files, can do it later after some of this settles down

Reply:
another thing, how would either approach work with Filterable.getFilter(CommandContext) ?

Reply:
Btw, I came to conclusion that packaging or adding reference not necessarily a good idea. Forgot to add comment to that effect.
Either way, not really needed here anyway.